Mage.Sets/src/mage/cards/t/TimotharBaronOfBats.java

```diff
@@ -0,0 +1,139 @@
+                                "It gains \"When this creature deals combat damage to a player, " +
+                                "sacrifice it and return the exiled card to the battlefield tapped\"."),
+                false,
+                anotherVampireFilter,
+                true
+        ));
+    }
+
+    private TimotharBaronOfBats(final TimotharBaronOfBats card) { super(card); }
+
+    @Override
+    public Card copy() { return new TimotharBaronOfBats(this); }
+}
+
+class TimotharBaronOfBatsCreateBatEffect extends OneShotEffect {
+
+    TimotharBaronOfBatsCreateBatEffect() {
+        super(Outcome.Benefit);
+        staticText = "exile it. " +
+                "If you do, create a 1/1 black Bat creature token with flying. " +
+                "It gains \"When this creature deals combat damage to a player, " +
+                "sacrifice it and return the exiled card to the battlefield tapped\".";
+    }
+
+    private TimotharBaronOfBatsCreateBatEffect(final TimotharBaronOfBatsCreateBatEffect effect) { super(effect); }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) { return false; }
+
+        Card card = game.getCard(targetPointer.getFirst(game, source));
+        if (card == null) { return false; }
+
+        player.moveCards(card, Zone.EXILED, source, game);
+
+        BatToken bat = new BatToken();
+        bat.addAbility(new DealsCombatDamageToAPlayerTriggeredAbility(
+                new TimotharBaronOfBatsReturnEffect(card, game),
+                false
+        ));
+
+        CreateTokenEffect createBatEffect = new CreateTokenEffect(bat, 1);
+        createBatEffect.setTargetPointer(new FixedTarget(card, game));
+        createBatEffect.apply(game, source);
+
+        return false;
+    }
+
+    @Override
+    public TimotharBaronOfBatsCreateBatEffect copy() { return new TimotharBaronOfBatsCreateBatEffect(this); }
+}
+
+class TimotharBaronOfBatsReturnEffect extends OneShotEffect {
+
+    private final MageObjectReference morOfCardToReturn;
+
+    TimotharBaronOfBatsReturnEffect(Card card, Game game) {
+        super(Outcome.PutCreatureInPlay);
+        staticText = "return the exiled card to the battlefield tapped";
+        this.morOfCardToReturn = new MageObjectReference(card, game);
+    }
+
+    private TimotharBaronOfBatsReturnEffect(final TimotharBaronOfBatsReturnEffect effect) {
+        super(effect);
+        this.morOfCardToReturn = effect.morOfCardToReturn;
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) { return false; }
+
+        Card card = morOfCardToReturn.getCard(game);
+        if (card == null) { return false; }
+
+        return player.moveCards(card, Zone.BATTLEFIELD, source, game, true, false, true, null);
```

Comment:
Doesn't look like you're sacrificing the bat anywhere in here "sacrifice it and return the exiled card to the battlefield tapped"

Reply:
Totally forgot about that, my bad.

Should I do this as a SacrificeSourceEffect().apply()?

Reply:
I think the best way would be to have the ability have 2 effects.

1. SacrificeSourceEffect
2. This custom effect.

Just use ability.addEffect before you add the ability to the bat.

Reply:
Alright, how's that

Reply:
You want the sacrifice effect to be first because of the way it's worded.  Then put "and" in the text of your custom effect so the text looks right in client when it triggers.

Reply:
fixed